src/api-service/__app__/onefuzzlib/pools.py

```diff
@@ -565,13 +584,28 @@ def cleanup_nodes(self) -> bool:
         nodes = Node.search_states(
             scaleset_id=self.scaleset_id, states=NodeState.ready_for_reset()
         )
-        if not nodes:
+
+        version = versions()
+        outdated = Node.search_outdated(
+            version=version["onefuzz"].version,
+            scaleset_id=self.scaleset_id, 
+            states=[NodeState.free]
+        )
+
+        if not (nodes or outdated):
             logging.debug("scaleset node gc done (no nodes) %s", self.scaleset_id)
             return False
 
+        to_stop = [] # need to send stop command to those nodes
```

Comment:
will be done when [this](https://github.com/microsoft/onefuzz/pull/32) is merged